marshal/tpms-types.c

```diff
@@ -0,0 +1,701 @@
+/***********************************************************************
+ * Copyright (c) 2015 - 2017, Intel Corporation
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ ***********************************************************************/
+
+#include <inttypes.h>
+#include <string.h>
+
+#include "sapi/marshal.h"
+#include "sapi/tpm20.h"
+#include "tss2_endian.h"
+#include "log.h"
+
+#define ADDR &
+#define VAL
+
+static TSS2_RC marshal_pcr_select(const UINT8 *ptr, uint8_t buffer[],
+                                  size_t buffer_size, size_t *offset)
+{
+    TPMS_PCR_SELECT *pcrSelect = (TPMS_PCR_SELECT *)ptr;
+    UINT32 i;
+    TSS2_RC ret;
+
+    if (ptr == NULL) {
+        LOG (WARNING, "src param is NULL");
+        return TSS2_TYPES_RC_BAD_REFERENCE;
+    }
+
+	ret = UINT8_Marshal(pcrSelect->sizeofSelect, buffer, buffer_size, offset);
```

Comment:
There are some weird whitespace errors at the front of a few lines here. These are probably left over from copy / paste from the old implementation.